packages/commonjs/src/base-cjs-module-system.ts

```diff
@@ -27,16 +27,31 @@ export interface IBaseModuleSystemOptions {
    * `undefined` - couldn't resolve request.
    */
   resolveFrom(contextPath: string, request: string, requestOrigin?: string): string | false | undefined;
+  /**
+   * Hook into file module evaluation.
+   */
+  loadModuleHook?: (loadModule: LoadModule) => LoadModule;
 }
 
-export function createBaseCjsModuleSystem(options: IBaseModuleSystemOptions): ICommonJsModuleSystem {
-  const { resolveFrom, dirname, readFileSync, globals = {} } = options;
+const falseModule = { exports: {}, filename: '', id: '', children: [] };
 
+export function createBaseCjsModuleSystem(options: IBaseModuleSystemOptions): ICommonJsModuleSystem {
+  const { resolveFrom, dirname, readFileSync, globals = {}, loadModuleHook } = options;
   const loadedModules = new Map<string, IModule>();
 
+  const load = loadModuleHook ? loadModuleHook(loadModule) : loadModule;
+
   return {
-    requireModule,
-    requireFrom,
+    requireModule(filePath) {
+      if (filePath === false) {
+        return {};
+      }
+      const fileModule = loadedModules.get(filePath) ?? load(filePath);
+      return fileModule.exports;
+    },
+    requireFrom(contextPath, request) {
+      return loadFrom(contextPath, request).exports;
```

Comment:
this may now return `falseModule.exports`, if `request` resolves to `false`. we want to check for `falseModule` and return a new object in that case.

Reply:
It's done inside

Reply:
```ts
    const resolvedPath = resolveThrow(contextPath, request, requestOrigin);
    if (resolvedPath === false) {
      return falseModule;
    }
```

Reply:
I see. We can eliminate all of these ifs if we use getter in the falseModule

Reply:
Everytime we get the export just return new object